homeassistant/components/nws_alerts/config_flow.py

```diff
@@ -0,0 +1,144 @@
+"""Config flow for NWS Alerts integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import requests
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import API_ENDPOINT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("api_key"): str,
+        vol.Required("friendly_name", default="NWS Alerts"): str,
+        vol.Required("update_interval", default=90): int,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict) -> dict:
+    """Validate the user input allows us to connect.
+
+    Return the user input (modified if necessary) or raise a vol.Invalid
+    exception if the data is incorrect.
+    """
+    endpoint = API_ENDPOINT.format(
+        lat=data["lat"], lon=data["lon"], api_key=data["api_key"]
```

Comment:
Import constants from `homeassistant.const`
```suggestion
        lat=data[CONF_LATITUDE], lon=data[CONF_LONGITUDE], api_key=data[CONF_API_KEY ]
```